x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/KqlAstBuilder.java

```diff
@@ -8,22 +8,222 @@
+        assert ctx.fieldName() != null; // Should not happen since the grammar does not allow the fieldname to be null.
+
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+        withFields(ctx.fieldName(), (fieldName, mappedFieldType) -> {
+            if (isRuntimeField(mappedFieldType) == false) {
+                boolQueryBuilder.should(QueryBuilders.existsQuery(fieldName));
+            }
+        });
+
+        return rewriteDisjunctionQuery(boolQueryBuilder);
+    }
+
+    @Override
+    public QueryBuilder visitRangeQuery(KqlBaseParser.RangeQueryContext ctx) {
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+
+        String queryText = ParserUtils.extractText(ctx.rangeQueryValue());
+        BiFunction<RangeQueryBuilder, String, RangeQueryBuilder> rangeOperation = rangeOperation(ctx.operator);
+
+        withFields(ctx.fieldName(), (fieldName, mappedFieldType) -> {
+            RangeQueryBuilder rangeQuery = rangeOperation.apply(QueryBuilders.rangeQuery(fieldName), queryText);
+            // TODO: add timezone for date fields
+            boolQueryBuilder.should(rangeQuery);
+        });
+
+        return rewriteDisjunctionQuery(boolQueryBuilder);
+    }
+
+    @Override
+    public QueryBuilder visitFieldLessQuery(KqlBaseParser.FieldLessQueryContext ctx) {
+        String queryText = ParserUtils.extractText(ctx.fieldQueryValue());
+
+        if (hasWildcard(ctx.fieldQueryValue())) {
+            // TODO: set default fields.
+            return QueryBuilders.queryStringQuery(escapeLuceneQueryString(queryText, true));
+        }
+
+        boolean isPhraseMatch = ctx.fieldQueryValue().QUOTED_STRING() != null;
+
+        return QueryBuilders.multiMatchQuery(queryText)
+            // TODO: add default fields?
+            .type(isPhraseMatch ? MultiMatchQueryBuilder.Type.PHRASE : MultiMatchQueryBuilder.Type.BEST_FIELDS)
+            .lenient(true);
+    }
+
+    @Override
+    public QueryBuilder visitFieldQuery(KqlBaseParser.FieldQueryContext ctx) {
+
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+        String queryText = ParserUtils.extractText(ctx.fieldQueryValue());
+        boolean hasWildcard = hasWildcard(ctx.fieldQueryValue());
+
+        withFields(ctx.fieldName(), (fieldName, mappedFieldType) -> {
+            QueryBuilder fieldQuery = null;
+
+            if (hasWildcard && isKeywordField(mappedFieldType)) {
+                fieldQuery = QueryBuilders.wildcardQuery(fieldName, queryText)
+                    .caseInsensitive(kqlParserExecutionContext.isCaseSensitive() == false);
+            } else if (hasWildcard) {
+                fieldQuery = QueryBuilders.queryStringQuery(escapeLuceneQueryString(queryText, true)).field(fieldName);
+            } else if (isDateField(mappedFieldType)) {
+                // TODO: add timezone
```

Comment:
⚠️ Will be exposed as a parameter of the kql query and done when introducing the query